### DISCUSSION TITLE
Update Teslamate

### DISCUSSION THREAD

**Discussion by Ramses7420** [teslamate-org/teslamate#3783] [General]
Hello, I want to update Teslamate, which is installed on my Mac. When I launch the command: Sudo Docker composes sweater, I have an error. I give you a screen copy. Thanks for your help

![SCR-20240326-oi6](https://github.com/teslamate-org/teslamate/assets/54696668/da10eb9f-9c58-4155-9b5d-7dcb04f09c67)

**Comment by cwanja** [comment]
Have you tried not using `sudo`? 
https://stackoverflow.com/questions/76590224/error-while-building-docker-container-using-linux-command-on-macos-must-use-a

**Reply by Ramses7420** [reply]
I just launched the order, but it's very long. Is it normal? And then, should I update Teslamate?
![SCR-20240326-oss](https://github.com/teslamate-org/teslamate/assets/54696668/b40facfc-51eb-4ba3-b497-ecd0df9b290f)

**Reply by cwanja** [reply]
It's the logging. You want to use the `-d` flag to disconnect in your `up` command.

**Reply by Ramses7420** [reply]
command like this : docker-compose -d

**Reply by cwanja** [reply]
`docker compose up -d`

https://docs.teslamate.org/docs/upgrading

Edit: if you are not using Docker Compose v2, use `docker-compose`

**Reply by Ramses7420** [reply]
Ok, and that doesn't risk anything, despite it continues to turn

**Reply by Ramses7420** [reply]
It didn't do anything

**Reply by cwanja** [reply]
Define 'risk'? It's just running detached from your terminal window and you can close the terminal window.

Learn more here: https://docs.docker.com/reference/cli/docker/compose/up/

**Reply by cwanja** [reply]
> It didn't do anything
> 
> 

What does that mean? Please provide logs and photos.

**Reply by Ramses7420** [reply]
I stay to block. That's what's going on
![SCR-20240326-p08](https://github.com/teslamate-org/teslamate/assets/54696668/9ccc91a9-c3f7-4a0c-aa4e-ef0ea27dc3fc)

**Reply by cwanja** [reply]
You need to bring the containers down first, run the upgrade command `pull` and then run the `up -d` commands. 

Please see the documentation.

**Comment by Ramses7420** [comment]
I closed the terminal window and I am always in 1.28.3
![SCR-20240326-p48](https://github.com/teslamate-org/teslamate/assets/54696668/7a2a5957-1dc8-4d94-8c8c-fa57f631910f)

**Comment by Ramses7420** [comment]
I'm really sorry to bother you, but what should I do exactly? I don't know very terminal under macOS

**Reply by cwanja** [reply]
Not a bother, just trying to keep the multiple replies threaded. 

Open Terminal. Go into the directory with your YML. Run `docker-compose down`. Then `docker-compose pull`. Then `docker-compose up -d`. 

If you face issues, please provide which step and screenshots and errors.

**Reply by Ramses7420** [reply]
Many thank's for your help/ It's ok. I'm on 1.28.5
